v2/federation/fetch.go

```diff
@@ -31,6 +33,16 @@ func WithSPIFFEAuth(bundleSource x509bundle.Source, endpointID spiffeid.ID) Fetc
 	})
 }
 
+// WithWebAuth authenticates the bundle endpoint with Web PKI authentication
+// using the provided X.509 root certificates instead of the system ones.
+func WithWebPKIAuth(rootCAs *x509.CertPool) FetchOption {
```

Comment:
We should probably fail if this is set along with `WithSPIFFEAuth`.

Also, maybe `WithWebPKIRoots`?